run_demo.sh

```diff
@@ -134,7 +134,7 @@ fi
 # address/IP-address.
 machine_hostname=$(hostname | tr '[:upper:]' '[:lower:]')
 if ! check_hostname "${machine_hostname}"; then
-  machine_hostname=$(ifconfig | grep 'inet ' | grep -v '127' | awk '{ print $2 }' | head -n 1 | cut -d '/' -f 1)
+  machine_hostname=$(ifconfig | grep 'inet ' | grep -v '127.\d.\d.\d' | awk '{ print $2 }' | head -n 1 | cut -d '/' -f 1)
```

Comment:
Does it work on your side?
I just made a local test on my machine with the following file named `test.txt`:

```txt
127.0.0.0
172.45.0.3
172.127.5.2
172.24.127.255
```

I get:

```bash
$ cat test.txt | grep -v '127.\d.\d.\d'
127.0.0.0
172.45.0.3
172.127.5.2
172.24.127.255
```

While this seems to work:

```bash
$ cat test.txt | grep -v '^127'
172.45.0.3
172.127.5.2
172.24.127.255
```


```suggestion
  machine_hostname=$(ifconfig | grep 'inet ' | grep -v '^127' | awk '{ print $2 }' | head -n 1 | cut -d '/' -f 1)
```

Reply:
I like this solution 👍 I've made the change and pushed it upstream

FYI also added a short print statement saying which 'hostname' is used for information purposes